.github/workflows/linuxjs-tests.yml

```diff
@@ -9,61 +9,42 @@ on:
     types: [published]
 
 jobs:
+  setup:
+    name: Setup test matrix
+    runs-on: ubuntu-latest
+    outputs:
+      group: ${{ steps.set-matrix.outputs.group }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Set matrix
+        id: set-matrix
+        shell: python
+        run: |
+          import json, os
+          from pathlib import Path
+
+          packages = []
+          for p in Path('packages').iterdir():
+              pkg_json = p / 'package.json'
+              if pkg_json.exists():
+                  data = json.loads(pkg_json.read_text())
+                  if data.get('scripts', {}).get('test:cov'):
+                      packages.append(f'js-{p.name}')
```

Comment:
Should we add some assertion that >1 package is found just so that it fails if for some reason it does not discover the packages?

Reply:
Done!